Brittany's Data Structures Exercises & CodingBat2

britth opened the pull request:
Completed the four data structure exercises from the ThinkPython book. Post can be found at the following link: http://britth.github.io/spring2014/2014/02/11/brittanydatastructures.html

britth committed:
Create 2014-02-11-brittanydatastructures.md

wrote extra credit data structures post, including code and reflections

britth committed:
Fixed headings

britth committed:
Fixed escape characters and spacing

britth committed:
Fixed line of code

britth committed:
removed one sentence from reflection

britth committed:
Fixed grammatical errors

gerbal commented:
This is a case where you probably don't need to put your code in includes, as this code won't run in a browser. 

If you put your comments in a multi-line comment block (`''' content '''`), and put it immediately after your `def function()`, python can automatically build documentation for that function using those comment blocks.

Otherwise, awesome post. It would be really helpful if you could include your books.txt in your pull request, that would make it easier for us to test your code.

britth committed:
Updated code and formatting

added multiline comment blocks to code for documentation. Added links to files used for my inputs. Fixed line of code in stringManipulationPG(file) function

britth committed:
Attempting to fix multi line comments

britth commented:
@gerbal Thanks for the tip! I can't figure out how to fix the multi-line comment blocks/documentation though, does that just apply if you put your code within the console and not code blocks? or is my markdown syntax wrong?

britth committed:
Merge pull request #3 from silshack/gh-pages

Merging in Recent Changes

gerbal commented:
Oh, the problem is just the indentation. the comment blocks should be indented the same as anything else following a `def function():`. See what I did here: http://hastebin.com/yorasubaja.py

eah13 commented:
@britth nice work!  I'll review this likely over the weekend but looks like you did a pretty good job with it.  We'll have some time to go over your answers (and anyone else who completed the EC) next week.

britth committed:
Fixing multi line comments

britth committed:
Created post for 2nd codingbat exercises

Created post with screenshots of 4 sections, three interesting exercises, and code/reflections for those exercises

britth committed:
Fixed some formatting, added more comments

britth committed:
Added additional comments

britth commented:
@eah13 sounds good! my codingbat 2 exercises are also completed and can be found at the following link: http://britth.github.io/spring2014/2014/02/12/brittanycodingbat2.html
Also, sorry if I missed this, but is there a way to separate pull requests, or is it okay for both posts to be in a single request? I just noticed all my codingbat 2 commits were added to this pull request.

gerbal commented:
You can make separate pull requests if you make a new branch. But you won't be able to preview your posts on github pages

eah13 commented:
We'll get to branches soon.  Mean time I've opened #256 to keep track of codingbat.

Excellent job with writeups and comments in your code!

eah13 commented on _posts/2014-02-12-brittanycodingbat2.md:
```diff
@@ -0,0 +1,112 @@
+###List-2: centered_average(nums)
+*Return the "centered" average of an array of ints, which we'll say is the mean average of the values, except ignoring the largest and smallest values in the array. If there are multiple copies of the smallest value, ignore just one copy, and likewise for the largest value. Use int division to produce the final average. You may assume that the array is length 3 or more.*
+
+There were a couple of exercises in the List-2 section that were a bit tricky for me, including this one. The big thing here was that I had trouble figuring out how to deal with cases where all of the values are the same. My code looked for the index of the max() and min() in order to not include those values in the avg; however, in cases where the max and min are the same, the index returned wou ...
```
Correctness is more important than efficiency!  ANd you got the correctness nailed.  In the real world you'll have many competing priorities for your time and you'll have to choose whether to refactor or not.  THere's plenty of production code (i.e. running in the world) that's inefficient.  As long as users don't notice it's probably OK!
